src/integration/java/org/springframework/data/couchbase/CouchbaseWaitStrategy.java

```diff
@@ -0,0 +1,131 @@
+package org.springframework.data.couchbase;
+
+import static java.time.temporal.ChronoUnit.SECONDS;
+
+import java.time.Duration;
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+
+import org.rnorth.ducttape.ratelimits.RateLimiterBuilder;
+import org.rnorth.ducttape.unreliables.Unreliables;
+import org.testcontainers.containers.Container;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.WaitStrategy;
+
+/**
+ * WaitStrategy for Couchbase containers which makes the Server node is
+ * initialized, RBAC user and default bucket is created.
+ */
+public class CouchbaseWaitStrategy implements WaitStrategy {
+
+	private Duration startupTimeout = Duration.of(60, SECONDS);
+	private Boolean rbacEnabled = false;
+
+	public CouchbaseWaitStrategy(String serverVerion) {
+		String[] verionSplit = serverVerion.split("\\.");
```

Comment:
you could use the Version thing from the SDK here

Reply:
The client bean is not available at this point.

Reply:
Yes thanks, fixed.